dclab/rtdc_dataset/fmt_hierarchy.py

```diff
@@ -284,6 +262,9 @@ def __getitem__(self, key):
         # to `self._events` in `self.apply_filter`.
         if key in self._events:
             return self._events[key]
+        elif len(self.hparent[key].shape) > 1:
+            self._events.setdefault(key, ChildNDArray(self, key))
```

Comment:
~This will create a `ChildNDArray` instance everytime this feature is accessed. I would suggest to add a case `if key not in self._events` and only then set it. In general `setdefault` would have been good though.~

EDIT: Just saw that you are iterating with `return self[key]`. This is not optimal because it adds another layer of complexity. Just set the key in `self._events` and return the value. In subsequent calls the first case will be used.

Reply:
Sounds good!

__Short sidenote__: AFAITIK (As far as **I thought** I knew) `setdefault` would only set the value to the key, if the key is not already used in the dictionary and not initiate the ChildNDArray object if the key is already existent.
And while the former is true, I did not consider that it would create a new ChildNDArray everytime the `setdefault`-function is called, which is actually the case.
Since the first `if`-statement already checks the existence of the key, the `setdefault`-function does not add anything and only confounds things here.